artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPConnectionContext.java

```diff
@@ -183,6 +188,11 @@ public void flush() {
    }
 
    public void close(ErrorCondition errorCondition) {
+      if (scheduledPool != null && scheduledPool instanceof ThreadPoolExecutor &&
```

Comment:
You can call cancel on the future to let the pool remote it and it would be better (just for likelihood with other similar declarations) to use the wildcard typed version of it or ScheduledFuture<?>

Reply:
The scheduledPool RemoveOnCancelPolicy is false so If I cancel the task it won't be removed from the queue and the garbage collector can't delete it.